hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/commit/HoodieMergeHelper.java

```diff
@@ -202,7 +202,9 @@ private Option<Function<HoodieRecord, HoodieRecord>> composeSchemaEvolutionTrans
       Schema newWriterSchema = AvroInternalSchemaConverter.convert(mergedSchema, writerSchema.getFullName());
       Schema writeSchemaFromFile = AvroInternalSchemaConverter.convert(writeInternalSchema, newWriterSchema.getFullName());
       boolean needToReWriteRecord = sameCols.size() != colNamesFromWriteSchema.size()
-          || SchemaCompatibility.checkReaderWriterCompatibility(newWriterSchema, writeSchemaFromFile).getType() == org.apache.avro.SchemaCompatibility.SchemaCompatibilityType.COMPATIBLE;
+          || !(SchemaCompatibility.checkReaderWriterCompatibility(newWriterSchema, writeSchemaFromFile).getType()
```

Comment:
I'm confused here.  If the schema is not compatible, the merging should fail here, correct?

Reply:
Can the incompatible avro schema be blent together?

Reply:
@danny0405 @yihua Thanks for your comment. What you said makes sense to me.  Is it reasonable to use `needToReWriteRecord = sameCols.size() != colNamesFromWriteSchema.size()
        &&  SchemaCompatibility.checkReaderWriterCompatibility(newWriterSchema, writeSchemaFromFile).getType()
        == SchemaCompatibility.SchemaCompatibilityType.COMPATIBLE;` ?

Reply:
This is the original logic right?

Reply:
> This is the original logic right?
The original logic  is
`boolean needToReWriteRecord = sameCols.size() != colNamesFromWriteSchema.size()
          || SchemaCompatibility.checkReaderWriterCompatibility(newWriterSchema, writeSchemaFromFile).getType() == org.apache.avro.SchemaCompatibility.SchemaCompatibilityType.COMPATIBLE;`

Reply:
> This is the original logic right?

The original logic is `boolean needToReWriteRecord = sameCols.size() != colNamesFromWriteSchema.size()
          || SchemaCompatibility.checkReaderWriterCompatibility(newWriterSchema, writeSchemaFromFile).getType() == org.apache.avro.SchemaCompatibility.SchemaCompatibilityType.COMPATIBLE;`

Reply:
So there is no need for rewrite if the schema is compatible and the fields number equals?

Reply:
I don't have much experience on this, @xiarixiaoyao do you have any suggestions?

Reply:
Alternatively, we could find another way to determine whether the newWriterSchema is consistent with the writeSchemaFromFile. If they are consistent, then there would be no need for rewriting.